.github/workflows/build.yml

```diff
@@ -22,46 +22,105 @@ env:
   RELEASE_OUT: "./release-out"
 
 jobs:
-  build:
+  test:
     runs-on: ubuntu-latest
     steps:
       -
         name: Checkout
         uses: actions/checkout@v3
-      -
-        name: Set up QEMU
-        uses: docker/setup-qemu-action@v2
       -
         name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v2
         with:
           version: latest
       -
         name: Test
-        run: |
-          make test
+        uses: docker/bake-action@v2
+        with:
+          targets: test
+          set: |
+            *.cache-from=type=gha,scope=test
+            *.cache-to=type=gha,scope=test
       -
-        name: Send to Codecov
+        name: Upload coverage
         uses: codecov/codecov-action@v3
         with:
           file: ./coverage/coverage.txt
+
+  prepare:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.platforms.outputs.matrix }}
+    steps:
       -
-        name: Expose GitHub Runtime
-        uses: crazy-max/ghaction-github-runtime@906832f62b7baa936e3fbef72b029308af505ee7
+        name: Checkout
+        uses: actions/checkout@v3
       -
-        name: Build binaries
+        name: Create matrix
+        id: platforms
         run: |
-          make release
-        env:
-          CACHE_FROM: type=gha,scope=release
-          CACHE_TO: type=gha,scope=release
+          echo ::set-output name=matrix::$(docker buildx bake binaries-cross --print | jq -cr '.target."binaries-cross".platforms')
```

Comment:
Matrix is dynamically created by looking at supported platforms in `docker-bake.hcl`